### PR TITLE
VSCODE-38: Send metrics when atlas link clicked

### PR DESCRIPTION
**Pull request by alenakhineika**
## Description

- Send metrics when "MongoDB Atlas" and "Create Free Cluster" links are clicked.
- Extract AST visiting methods to a separate Visitor class to improve code readability.
- Mock long-running functions in completion tests.

### Checklist
- [x] New tests and/or benchmarks are included
- [ ] Documentation is changed or added

## Motivation and Context

- [ ] Bugfix
- [x] New feature
- [ ] Dependency update
- [ ] Misc

## Types of changes

- [ ] *Backport Needed*
- [ ] Patch (non-breaking change which fixes an issue)
- [x] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: src/views/webview-app/extension-app-message-constants.ts [hunk 1]
```diff
@@ -5,7 +5,8 @@ export enum MESSAGE_TYPES {
   CONNECT_RESULT = 'CONNECT_RESULT',
   OPEN_CONNECTION_STRING_INPUT = 'OPEN_CONNECTION_STRING_INPUT',
   OPEN_FILE_PICKER = 'OPEN_FILE_PICKER',
-  FILE_PICKER_RESULTS = 'FILE_PICKER_RESULTS'
+  FILE_PICKER_RESULTS = 'FILE_PICKER_RESULTS',
+  LINK_CLICKED = 'LINK_CLICKED'
```
Comment (Anemy) [inline src/views/webview-app/extension-app-message-constants.ts:9]: I think we can make this `LINK_CLICKED` action a bit more descriptive, maybe `CREATE_ATLAS_CLUSTER_LINK_CLICKED`

File: src/telemetry/telemetryController.ts [hunk 2]
```diff
@@ -8,7 +8,8 @@ import { StorageController } from '../storage';
 const log = createLogger('analytics');
 
 export enum TelemetryEventTypes {
-  PLAYGROUND_CODE_EXECUTED = 'playground code executed'
+  PLAYGROUND_CODE_EXECUTED = 'playground code executed',
+  LINK_CLICKED = 'link clicked'
```
Comment (Anemy) [inline src/telemetry/telemetryController.ts:12]: I think we can make this `LINK_CLICKED` action a bit more descriptive, maybe `CREATE_ATLAS_CLUSTER_LINK_CLICKED`? - Also maybe for the action inside the webview.

File: src/views/webview-app/extension-app-message-constants.ts (hunk 1 above)
Comment (alenakhineika) [reply src/views/webview-app/extension-app-message-constants.ts:9]: We are going to have one metric `link clicked` for any link clicking. For now, it is the Atlas link, but later can be anything.

For example, now we have two links actually that send these metrics:
`{event: 'link clicked', properties: {screen: 'connectScreen', linkId: 'atlasLanding'}}`
`{event: 'link clicked', properties: {screen: 'connectScreen', linkId: 'freeClusterCTA'}}`

So I thought that would be nice to have one action and reuse it for any link. What do you think?

Comment (Anemy) [reply src/views/webview-app/extension-app-message-constants.ts:9]: ahhh. That makes sense. An idea we could do is in order to ensure these links come in that format is to add a type for tracking events. So in the telemetry controller we could change `public track(eventType: TelemetryEventTypes, properties: object): void { ... }` to 

```typescript
type PlaygroundTelemetryEventProperties = {
  type: string;
};

type LinkClickedTelemetryEventProperties = {
  screen: string;
  linkId: string;
};

type TelemetryEventProperties = PlaygroundEventProperties | LinkClickedTelemetryEventProperties;

public track(eventType: TelemetryEventTypes, properties: TelemetryEventProperties): void { ... }
```
Maybe this is a bit overtyping, but a suggestion.

Comment (alenakhineika) [reply src/views/webview-app/extension-app-message-constants.ts:9]: I like it a lot!! Pushed changes. Thx for suggesting ✨

**Pull request merged**